### ISSUE TITLE
Move Locking Mechanism to Wrestic

### ISSUE THREAD

**Issue by tobru** [vshn/k8up#324]
## Summary

**As** the K8up Operator
**I want** to get rid of the locking code
**So that** things get less complicated.

## Context

Currently the K8up operator contains a rather complicated locking mechanism, which is also prone to some bugs. This locking mechanism is legacy for when all backups were made to a single bucket, instead of different ones.

It's also somewhat redundant as Restic also contains a locking mechanism. This mechanism can be used to replace K8ups own logic:

* wrestic checks the lock state before starting each run
* depending on the state it will either wait for the lock to be released or it will simply start to run the job
* if it needs to wait it will periodically pull the lock state from the remote.

This will greatly simplify the logic used in the Operator, as well as avoid some hard to debug bugs.

The Prometheus metrics will have to be moved, too. As they are increased and decreased on locking and unlocking events.

## Out of Scope

* -

## Further links

* -

**Comment by Kidswiss** [comment]
There's already some waiting logic in Wrestic: https://github.com/vshn/wrestic/blob/80a456e92aac0baf2abbd0fa5a7c14153012e97b/restic/wait.go#L22

HOWEVER: I found that restic's own locking is not really perfect and also contains some race-conditions. I use a similar approach for non-k8s backups and there it sometimes fails to determine the right locks.

**Comment by Kidswiss** [comment]
Additional brain dump: with the idea going around that k8up could also get a client mode we could then move the existing locking logic to there.

**Comment by ccremer** [comment]
Since we plan to integrate wrestic in K8up later on, this issue can be closed
